### ISSUE TITLE
Automate unit testing

### ISSUE THREAD

**Issue by dprice000** [dprice809/LazyPI#7]
The project is getting to a point where many pieces are completed. Some tests have been created, but there is no automated server at this time. 

To add some integrity to the library, setting up an automated test would be ideal. Sadly there are not a lot of OpenSource solutions for this. Also, I do not have a good test machine for the WebAPI. The best way to test this library is to run it against the WebAPI.

**Comment by pthivierge** [comment]
Hello David,  I have seen one of our team using https://www.appveyor.com/ at some point in time. It does continuous build and tests.  if you are lucky that will work seamleassly with your project.  If not, that will require a small tweaks.  Everything works based on the github repository, it is convenient. 

On top right corner, under your avatar menu, select integration.  You should see appVeyor in there.

This should give you a quick start, there maybe other similar applications thoug that I am not ware of :)

**Comment by dprice809** [comment]
AppVeyor looks like one of the best tools for OpenSource C# projects. One of the issues I have is that I do not have a designated test WebAPI system. I have been lucky enough to test it against live WebAPI setups, but I can not use those systems as a designated test server.

**Comment by dprice809** [comment]
I'm realizing now that I can probably create Test versions of each of the Controller objects. From there I can just fake the WebAPI Interactions.

**Comment by pthivierge** [comment]
Yes, that would work, this takes time though and needs to be maintained over time.  I am wondering if there are tools to generate test objects from APIs. I just looked and I was not successful.  

We are also looking into having a publicly available PI Web API, we are not there yet though ;)
